boa/code/compiler.py

```diff
@@ -0,0 +1,40 @@
+import abc
+import os
+
+
+class ICompiler(abc.ABC):
+    @abc.abstractmethod
+    def save(self, compiler, path, output_path=None):
```

Comment:
I think this interface should explicitly define the argument types and a return type
The Neo3 implementation should throw an "Not implemented exception"